storage/src/lib.rs

```diff
@@ -191,6 +200,19 @@ impl Storage {
         Err(Error::BlockNotFound(hash.clone()))
     }
 
+    pub fn block_location_by_height(&self, height: u64) -> Result<BlockLocation> {
+        info!("Lokup size from request: {}", self.lookups.len());
+        let mut h = height;
+        for (packref, lookup) in self.lookups.iter() {
+            let indexfile::FanoutTotal(total) = lookup.fanout.get_total();
+            if h < (total as u64) {
+                return Ok(BlockLocation::Packed(packref.clone(), h as u32, indexfile::IndexOffsetType::Unsorted));
+            }
+            h -= total as u64;
+        }
+        Err(Error::BlockHeightNotFound(height))
+    }
```

Comment:
This is the core logic for the block-by-height query search. Super simple, but for this to work - we need to:
1. Have lookups to be ordered in the chain order
2. Have block offsets available in the chain order for each epoch